### PR TITLE
Do not include ssl_cert and ssl_key if values are nil

### PR DESCRIPTION
**Pull request by hollistermala**

### CODE REVIEW

**Review by ramereth — CHANGES_REQUESTED** [review]
Can you please at least add unit tests which ensure this works properly when it is set to null?

File: test/cookbooks/osl-imap-test/recipes/no_ssl.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+include_recipe 'osl-imap-test::default'
+
+osl_imap_dovecot 'default' do
+  auth_type :system
+  ssl_cert nil
+  ssl_key nil
```
Comment (ramereth) [inline test/cookbooks/osl-imap-test/recipes/no_ssl.rb]: I believe these should be nil by default. Can you please test this without setting them nil? Also, can you add a test suite to kitchen and some integration tests to ensure this works properly?

**Review by ramereth — CHANGES_REQUESTED** [review]

**Review by ramereth — APPROVED** [review]
Minor bump

**Comment by hollistermala** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v3.1.0.  Have a nice day!
